src/Jackett.Common/Utils/Extensions.cs

```diff
@@ -41,9 +41,13 @@ public static class EnumerableExtension
     {
         public static string AsString(this IEnumerable<char> chars) => string.Concat(chars);
 
-        // Should be collection.Any()
-        // Remove in favor of existing built in function?
-        public static bool IsEmpty<T>(this IEnumerable<T> collection) => collection.Count() > 0;
+        public static T FirstIfSingleOrDefault<T>(this IEnumerable<T> enumerable, T replace = default)
+        {
+            //Avoid enumerating the whole array.
+            //If enumerable.Count() < 2, takes whole array.
+            var test = enumerable.Take(2).ToList();
```

Comment:
All Linq functions have internal logic to optimize different types of underlying types. In this case Take, ToList, and Count() have optimizations if the underlying collection is of type IList. It would be even better if they publicly exposed `GetCount(bool onlyIfCheap)`

Reading the Net-Core source to see those optimizations, Line 48 under the hood looks like (in the case of IList):
```C#
var test = new List<T>()
for(int i = 0; i < 2; i++)
test.Add(orignal[i]);
```

There is one optimization we can do in this case though, by adding:
```C#
if(enumerable is IList list)
    return list.Count == 1 ? list[0] : replace;
```

Reply:
Running a profile in the C# interactive console, on a loop of 10,000 lists with random number of string elements, it's basically a wash.
- Not optimized: 00.0297677; 00.0303638; 00.0340765;
- Optimized: 00.0213106; 00.0272700; 00.0501906;

Reply:
This method is used only 7 times in Lists with 0, 1 or 2 values. As a learning exercise is fine but it won't make a difference in performance. Do what you think is best for the project.